server/api/discord.ts

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+
+import {
+  clientId,
+  clientSecret,
+  REDIRECT_URI,
+  BASE_DISCORD_API_URL
+} from '../src/config';
+
+
+
+export class Discord {
+
+  /** Axios request builder */
+  static async request(endpoint: string, headers: any, body = {}, method = "GET") {
+    const url = `${BASE_DISCORD_API_URL}/${endpoint}`;
+    const data = (method === "GET")
+      ? undefined
+      : body;
```

Comment:
the default values for `body` and `method` are not in sync for the same case (either GET or POST)

otherwise, it seems like this could this simplified to:
```suggestion
  static async request(endpoint: string, headers: any, data = undefined, method = "GET") {
    const url = `${BASE_DISCORD_API_URL}/${endpoint}`;
```

Reply:
Hmm not sure why it didn't work when I first tried it last time, but made the suggested change now and it works. Probably had a typing issue before that I didn't catch earlier.